addons/account_edi/models/account_move.py

```diff
@@ -0,0 +1,117 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, models
+from odoo.tools.pdf import OdooPdfFileReader
+
+from lxml import etree
+
+import io
+import base64
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    @api.returns('mail.message', lambda value: value.id)
+    def message_post(self, **kwargs):
+        # OVERRIDE
+        # /!\ 'default_res_id' in self._context is used to not process attachment when using a form view.
+        res = super(AccountMove, self).message_post(**kwargs)
+
+        if len(self) != 1 or self.env.context.get('no_new_invoice') or not self.is_invoice(include_receipts=True):
+            return res
+
+        for attachment in self.env['ir.attachment'].browse(kwargs.get('attachment_ids', [])):
+            self._create_invoice_from_attachment(attachment)
+
+        return res
+
+    def _create_invoice_from_attachment(self, attachment):
+        ''' Generic method decoding the ir.attachment record passed as parameter in order to create an invoice.
+        :param attachment:  An ir.attachment record.
+        '''
+        self.ensure_one()
+        if 'pdf' in attachment.mimetype:
+            self._create_invoice_from_pdf(attachment)
+        if 'xml' in attachment.mimetype:
+            self._create_invoice_from_xml_attachment(attachment)
```

Comment:
this part is confusing...
`_create_invoice_from_pdf` is creating an invoice from a pdf but... what is `self` then? I don't understand...